components/function-controller/internal/controllers/knative/service_controller.go

```diff
@@ -68,6 +69,11 @@ func (r *ServiceReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !hasCorrectLabels(*instance) {
+		r.Log.WithValues("service", request.NamespacedName).Info("skipping reconcilation for a service without needed labels")
+		return ctrl.Result{}, nil
+	}
+
 	if !instance.Status.IsReady() {
 		return ctrl.Result{RequeueAfter: r.config.RequeueDuration}, nil
 	}
```

Comment:
I know it is not related to this PR, but it doesn't make sense to add here `RequeueAfter` as on Service Instance status change controller will get update event.

`RequeueAfter` should be a default result from `Reconcile` method when all operations finished sucessufull, or no operation needed (revisions are cleaned)

Reply:
ok, I will delete it

now I'm wondering whether we need `requeueAfter` at all, as we operate on events

Reply:
As a default return from `Reconcile` method